File: leetcode/1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        r = 0
        i = 0
        for c in s:
            if c == 'R':
                i += 1
            else:
                i -= 1
            if i == 0:
                r += 1
        return r

File: leetcode/1221.c
int balancedStringSplit(char * s){
    int r = 0, i = 0;
    for(; *s; s++)
    {
        if(*s == 'L') {
            i ++;
        } else {
            i --;
        }
        if(i == 0) {
            r ++;
        }
    }
    return r;
}
